vscode_constants.py

```
@@ -12,6 +15,33 @@ def get_bin_path(bin_dir):
+        # No exec bit on Windows; Popen resolves the suffix-less path to .exe itself.
+        return os.path.isfile(path) or os.path.isfile(path + ".exe")
+    if not os.path.isfile(path):
+        logger.info("No runnable binary at %s; keeping the bare command for PATH lookup", path)
+        return False
+    if os.access(path, os.X_OK):
+        return True
+    if os.path.islink(path):
+        logger.warning("%s is a symlink without exec permission; refusing to chmod it", path)
+        return False
+    try:
+        os.chmod(path, 0o755)
```

Ruff (0.15.15)

[error] `os.chmod` setting a permissive mask `0o755` on file or directory

(S103)